jina/hubble/hubio.py

```diff
@@ -113,10 +148,110 @@ def push(self) -> None:
                     + colored(usage, 'cyan', attrs='underline')
                 )
 
-            else:
+            elif resp.text:
+                # NOTE: sometimes resp.text returns empty
                 raise Exception(resp.text)
+            else:
+                resp.raise_for_status()
 
         except Exception as e:  # IO related errors
             self.logger.error(
                 f'Error when trying to push the executor at {self.args.path}: {e!r}'
             )
+
+    def fetch(self, id: str, tag: Optional[str] = None) -> HubExecutor:
+        """Fetch the executor meta info from Jina Hub.
+        :param id: the ID of the executor
+        :param tag: the version tag of the executor
+        :return: meta of executor
+        """
+        with ImportExtensions(required=True):
+            import requests
+
+        pull_url = JINA_HUBBLE_PUSHPULL_URL + f'/{id}'
+        path_params = []
+        if self.args.secret:
+            path_params.append(f'secret={self.args.secret}')
+        if tag:
+            path_params.append(f'tag={tag}')
+
+        if len(path_params) > 0:
+            pull_url += '?' + ','.join(path_params)
```

Comment:
hmm, this looks very fragile as it does not escape special characters such as `?`, `&`, `/`

please refer to https://stackoverflow.com/a/10765733